src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java

```diff
@@ -24,248 +20,290 @@
+        File file = new File(path);
+        Image image = new Image(file.toURI().toString());
+        ImageView iv = new ImageView();
+        iv.setPreserveRatio(true);
+        iv.setFitHeight(20);
+        iv.setImage(image);
+        iv.visibleProperty().bind(visible);
+
+        return iv;
+    }
+
+    public void setModel(IElevatorWrapper model) {
+        this.model = model;
+
+        //Init elevator floors
+        try {
+            info = model.queryGeneralInformation();
+        } catch (Exception e) {
+            errorText.concat(e.getMessage() + "\n");
+        }
+
+        floorNames.clear();
+        gElevatorFloors.getChildren().clear();
+        floors = new FloorState[info.getNrOfFloors()];
+
+        lTopFloor.setText(((Integer) (info.getNrOfFloors() - 1)).toString());
+
+        for (int i = 0; i < info.getNrOfFloors(); i++) {
+            floorNames.add("Floor " + i);
+            floors[i] = new FloorState();
+
+            RowConstraints rCon = new RowConstraints();
+            rCon.setMinHeight(20);
+            rCon.setPercentHeight(100.0 / info.getNrOfFloors());
+
+            HBox hb = new HBox();
+            hb.setAlignment(Pos.CENTER_RIGHT);
+            hb.setSpacing(5);
+            hb.setPadding(new Insets(5, 5, 5, 5));
+            hb.translateXProperty().set(10);
+
+            Line line = new Line(0, 0, 10, 0);
+            line.disableProperty().bind(floors[i].isServiced.not());
+
+            Label label = new Label(Integer.toString(i));
+            label.disableProperty().bind(floors[i].isServiced.not());
+
+            hb.getChildren().addAll(
+                    createFloorImageView("images/arrowUp.png", floors[i].requestUp),
+                    createFloorImageView("images/arrowDown.png", floors[i].requestDown),
+                    createFloorImageView("images/hand.png", floors[i].stopRequest),
+                    createFloorImageView("images/filler.jpeg", targetFloor.isEqualTo(i)),
+                    label,
+                    line);
+
+            gElevatorFloors.getRowConstraints().add(rCon);
+            gElevatorFloors.add(hb, 0, info.getNrOfFloors() - i - 1);
+        }
+
+        elevators.clear();
+        for (int i = 0; i < info.getNrOfElevators(); i++) {
+            elevators.add("Elevator " + i);
+        }
+
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                update();
+            }
+        }, 0, 100);
+    }
+
+    private void update() {
+        if (model == null) {
+            errorText.set(errorText.get() + "Invalid model\n");
+            return;
+        } else if (currentElevator.get() < 0)
+            return;
+
+        Platform.runLater(() -> {
+            ElevatorState elevatorState;
+
+            try {
+                elevatorState = model.queryElevatorState(currentElevator.intValue());
+            } catch (Exception e) {
+                errorText.set(errorText.get() + e.getMessage() + "\n");
```

Comment:
Above you used `errorText.concat`. Does that work here too?

Reply:
Separation of concerns: The exception handling code should not need to know how to display errors to the user (but just how to cope with the exception itself). Instead, move that code into a separate method (e.g. `log(String message)`, or `log(Exception e)`, or both) and just call this method wherever needed.

Reply:
>Above you used errorText.concat. Does that work here too?

`concat` doesn't work because it returns a new string or in this case a new binding to a concated string

> Separation of concerns

Resolved by using a logging function